blender/LilySurfaceScraper/Scrapers/LocalDirectoryScraper.py

```diff
@@ -21,75 +21,121 @@
 # This file is part of LilySurfaceScraper, a Blender add-on to import materials
 # from a single URL
 
-import zipfile
 import os
-from os import path
-import re
-from urllib.parse import urlparse, parse_qs, urlencode, urljoin
 from .AbstractScraper import AbstractScraper
 
+
 class LocalDirectoryScraper(AbstractScraper):
     """
     This scraper does not actually scrap a website, it links textures from a
     local directory, trying to guess the meaning of textures from their names.
     """
     source_name = "Local Directory"
+    scraped_type = {'MATERIAL', "WORLD", "LIGHT"}
     home_url = None
+    show_preview = False
 
     _texture_cache = None
 
     @classmethod
     def canHandleUrl(cls, url):
         """Return true if the URL can be scraped by this scraper."""
-        return path.isdir(url)
+        return os.path.isdir(url) or os.path.isfile(url)
 
-    def fetchVariantList(self, url):
+    def fetchVariantList(self, path):
```

Comment:
I though the split between fetchVariantList and getVariantList was that the latter is the "virtual" one (as in C++, meaning the one meant to be overridden in subclasses) but not the former. And here despite being in a subclass we change fetchVariantList? I'm not sure I got which is supposed to do what exactly then. ;)

(Also, I usually prefix with "fetch" functions that query http resources to insist on the fact that it is costly, whereas "get" means to me that it accesses something in local memory.

Reply:
ScrapaedData calls `fetchVariantList` that then calls `getVariantList` (which gets implemented by subclasses) and saves the metadata it got. But in the case of the LocalDirectoryScraper, it does not need to save the metadata or download thumbnails, so it directly overrides `fetchVariantList`.